rdf-abac-fuseki/src/main/java/io/telicent/jena/abac/fuseki/server/UserInfoEnrichmentFilter.java

```diff
@@ -0,0 +1,457 @@
+     * <p>Converts a structured {@link JsonObject} into a list of {@code "k=v"} strings, where:
+     * <ul>
+     *   <li>arrays become repeated {@code key=value} entries,</li>
+     *   <li>objects become {@code key.sub=value},</li>
+     *   <li>scalars become {@code key=value}.</li>
+     * </ul>
+     *
+     * @param v a JSON value expected to be an object
+     * @return a list of {@code "k=v"} strings, or {@code null} if {@code v} is not an object
+     */
+    @SuppressWarnings("unused")
+    protected static List<String> flattenFromAttributesObject(JsonValue v) {
+        if (v == null || !v.isObject()) return null;
+        List<String> out = new ArrayList<>();
+        JsonObject obj = v.getAsObject();
+        for (String key : obj.keys()) {
+            JsonValue val = obj.get(key);
+            if (val == null) continue;
+            if (val.isArray()) {
+                JsonArray a = val.getAsArray();
+                a.forEach(x -> {
+                    String s = jsonScalarAsString(x);
+                    if (s != null) {
+                        out.add(key + "=" + s);
+                    }
+                });
+            } else if (val.isObject()) {
+                JsonObject m = val.getAsObject();
+                for (String k : m.keys()) {
+                    String s = jsonScalarAsString(m.get(k));
+                    if (s != null) {
+                        out.add(key + "." + k + "=" + s);
+                    }
+                }
+            } else {
+                String s = jsonScalarAsString(val);
+                if (s != null) {
+                    out.add(key + "=" + s);
+                }
+            }
+        }
+        return out;
+    }
+
+    /**
+     * Build an {@link AttributeValueSet} from either:
+     * <ul>
+     *   <li>a {@link JsonArray} of {@code "k=v"} strings, or</li>
+     *   <li>a {@link JsonObject} treated as a structured set of attributes and flattened to {@code "k=v"}.</li>
+     * </ul>
+     * Non-conforming entries are ignored.
+     *
+     * @param v the JSON value to interpret
+     * @return a non-null {@link AttributeValueSet} (possibly empty)
+     */
+    static AttributeValueSet flattenToAttributeValueSet(JsonValue v) {
+        if (v == null) return AttributeValueSet.EMPTY;
+
+        List<AttributeValue> out = new ArrayList<>();
+
+        if (v.isArray()) {
+            // Expect an array of "k=v" strings
+            JsonArray a = v.getAsArray();
+            a.forEach(x -> {
+                if (x != null && x.isString()) {
+                    String s = x.getAsString().value();
+                    if (s != null && !s.isBlank()) {
+                        AttributeValue value = parseValue(s);
+                        if (value != null) {
+                            out.add(value);
+                        }
+                    }
+                }
+            });
+            return AttributeValueSet.of(out);
+        }
+
+        if (v.isObject()) {
```

Comment:
```suggestion
        else if (v.isObject()) {
```

Since the two branches are mutually exclusive